### PR TITLE
deploy: set gracePeriodSeconds on deployer deletion

### PR DESCRIPTION
**Pull request by kargakis**
Ref https://github.com/openshift/origin/issues/9681

### CODE REVIEW

**Comment by kargakis** [comment]
[test]

**Comment by smarterclayton** [comment]
I don't know that this is what you want.  How is this helping?  You're not giving the deployer a chance to signal other items.

**Comment by kargakis** [comment]
> I don't know that this is what you want. How is this helping? You're not giving the deployer a chance to signal other items.

What do you mean by "signal other items"? Isn't the [delete event handler for pods](https://github.com/openshift/origin/blob/4db1957ef87707bff37d1065b7172aace8c080de/pkg/deploy/controller/deployment/factory.go#L134-152) going to work (and signal the deployment to be reconciled again) ?

**Comment by smarterclayton** [comment]
If this is fixing your problem then you are probably hitting another
problem.

On Tue, Jul 12, 2016 at 12:33 PM, Michail Kargakis <notifications@github.com

**Comment by smarterclayton** [comment]
All this is probably doing is avoiding whatever race is in the deployment
controller because you don't get a second update to the pod.  We just need
to find the race.

**Comment by kargakis** [comment]
I can't find anything suspicious in the deployment controller. Note that the test uses a very ambitious timeout of 10 seconds. Updating both the timeout and gracePeriodSeconds of the deployer pod

**Comment by smarterclayton** [comment]
The 10s test timeout was probably too much, agree.

**Comment by kargakis** [comment]
Assuming LGTY [merge]

File: pkg/deploy/controller/deployment/controller.go [hunk 1]
```diff
@@ -369,8 +369,9 @@ func (c *DeploymentController) cleanupDeployerPods(deployment *kapi.ReplicationC
 	}
 
 	cleanedAll := true
+	gracePeriod := int64(10)
 	for _, deployerPod := range deployerList.Items {
-		if err := c.pn.Pods(deployerPod.Namespace).Delete(deployerPod.Name, &kapi.DeleteOptions{}); err != nil && !kerrors.IsNotFound(err) {
+		if err := c.pn.Pods(deployerPod.Namespace).Delete(deployerPod.Name, &kapi.DeleteOptions{GracePeriodSeconds: &gracePeriod}); err != nil && !kerrors.IsNotFound(err) {
```
Comment (smarterclayton) [inline pkg/deploy/controller/deployment/controller.go]: Actually, the correct place to put this is where you create the deployer pod, not here.

**Pull request merged**